clients/context-registry/src/ContextRegistryClient.ts

```diff
@@ -115,122 +106,96 @@ export class Project extends CommonAssetProjectContext {
-   * @returns Resolves to the found asset. Rejects if no assets, or more than one asset is found.
+   * @param arg Options for paging and/or searching
+   * @returns Array of iTwins, may be empty
    */
-  public async getAsset(requestContext: AuthorizedClientRequestContext, queryOptions?: RequestQueryOptions): Promise<Asset> {
-    requestContext.enter();
-    const assets: Asset[] = await this.getAssets(requestContext, queryOptions);
-    requestContext.enter();
-    if (assets.length === 0)
-      throw new Error("Could not find an asset with the specified criteria that the user has access to");
-    else if (assets.length > 1)
-      throw new Error("More than one asset found with the specified criteria");
-
-    return assets[0];
+  public async getAll(requestContext: AuthorizedClientRequestContext, arg?: ITwinQueryArg): Promise<ITwin[]> {
+    const queryOptions: RequestQueryOptions = {
+      $top: arg?.pagination?.top,
+      $skip: arg?.pagination?.skip,
+    };
+
+    if (arg?.search) {
+      if (arg.search.exactMatch)
+        queryOptions.$filter = `${arg.search.propertyName}+eq+'${arg.search.searchString}'`;
+      else
+        queryOptions.$filter = `${arg.search.propertyName}+like+'${arg.search.searchString}'`;
+    }
+
+    return this.getByQuery(requestContext, queryOptions);
   }
 
-  /** Gets the iTwin asset contexts that are accessible to the authorized user.
+  /** Gets all iTwins (projects or assets) using the given query options
    * @param requestContext The client request context
-   * @param queryOptions Query options. Use the mapped EC property names in the query strings and not the TypeScript property names.
-   * @returns Resolves to an array of assets.
+   * @param queryOptions Use the mapped EC property names in the query strings and not the TypeScript property names.
+   * @returns Array of iTwins meeting the query's requirements
    */
-  public async getAssets(requestContext: AuthorizedClientRequestContext, queryOptions?: RequestQueryOptions): Promise<Asset[]> {
+  private async getByQuery(requestContext: AuthorizedClientRequestContext, queryOptions?: RequestQueryOptions): Promise<ITwin[]> {
     requestContext.enter();
-    return this.getInstances<Asset>(requestContext, Asset, "/Repositories/BentleyCONNECT--Main/ConnectedContext/Asset", queryOptions);
+    const projectQuery = queryOptions;
+    const assetQuery = queryOptions;
+
+    if (queryOptions?.$skip && (projectQuery && assetQuery)) {
+      // Ceiling to skip a project object on Odd skips
+      projectQuery.$skip = queryOptions?.$skip ? Math.ceil(queryOptions.$skip / 2) : undefined;
+
+      // Floor to skip an asset object on Even skips
+      assetQuery.$skip = queryOptions?.$skip ? Math.floor(queryOptions.$skip / 2) : undefined;
+    }
+
+    const projectITwins: ITwin[] = await this.getInstances<Project>(requestContext, Project, "/Repositories/BentleyCONNECT--Main/ConnectedContext/project/", projectQuery);
```

Comment:
Are there plans to switch to the public endpoints ? If so, would it be a good time to change this ?

Reply:
Yes that is the plan. The reason it is still hitting these endpoints is because the CONNECT team is in the process of adding a more generic ITwin endpoint but so far they are only exposing projects. Once they have a public API that covers more than just projects I will switch it over.